src/agents/workflows/subgraphs/step2_router/graph.py

```
@@ -253,6 +288,38 @@ def intent(state: AgentState) -> AgentState:
+    if _is_next_problem_request(latest_question):
+        indexed_problems = state.get("problems") or []
+        if isinstance(indexed_problems, list):
+            current_index = int(state.get("current_problem_index", 0) or 0)
+            if 0 <= current_index + 1 < len(indexed_problems):
+                state["current_problem_index"] = current_index + 1
+                state["plan"] = ["Solve"]
+                state["simple_response"] = False
+                state["prev_action"] = "Intent"
+                return state
+
     latest_question, ocr_full_text, combined_question, conversation_context = (
```

Ruff (0.15.0)

[warning] Unpacked variable `ocr_full_text` is never used

Prefix it with an underscore or any other dummy variable pattern

(RUF059)